File: CustomQueries/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs
using Reveal.Sdk;
using Reveal.Sdk.Data;
using Reveal.Sdk.Data.Microsoft.SqlServer;
using Reveal.Sdk.Data.MySql;

namespace RevealSdk.Server.Reveal
{
    public class DataSourceProvider : IRVDataSourceProvider
    {
        public Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId,
            RVDataSourceItem dataSourceItem)
        {
            //update underlying data source
            ChangeDataSourceAsync(userContext, dataSourceItem.DataSource);

            if (dataSourceItem is RVSqlServerDataSourceItem sqlDataSourceItem)
            {
                if (sqlDataSourceItem.Id == "MySqlServerDataSourceItem")
                {
                    //get the sales-person-id from the userContext
                    var salesPersonId = userContext.Properties["sales-person-id"];

                    //parametrize your custom query with the property obtained before
                    sqlDataSourceItem.CustomQuery =
                        $"SELECT * FROM [Sales].[SalesOrderHeader] WHERE [SalesPersonId] = {salesPersonId}";
                }
            }

            if (dataSourceItem is RVMySqlDataSourceItem mySqlDataSourceItem)
            {
                if (mySqlDataSourceItem.Id == "MyMySqlDataSourceItem")
                {
                    //get the sales-person-id from the userContext
                    var salesPersonId = userContext.Properties["sales-person-id"];

                    //parametrize your custom query with the property obtained before
                    mySqlDataSourceItem.CustomQuery =
                        $"SELECT * FROM Sales_SalesOrderHeader WHERE SalesPersonId = {salesPersonId}";
                }
            }

            return Task.FromResult(dataSourceItem);
        }

        public Task<RVDashboardDataSource> ChangeDataSourceAsync(IRVUserContext userContext,
            RVDashboardDataSource dataSource)
        {
            if (dataSource is RVSqlServerDataSource sqlDataSource)
            {
                sqlDataSource.Host = "your-host";
                sqlDataSource.Database = "your-database";
            }

            if (dataSource is RVMySqlDataSource mySqlDataSource)
            {
                mySqlDataSource.Host = "your-host";
                mySqlDataSource.Database = "your-database";
            }

            return Task.FromResult(dataSource);
        }
    }
}

File: CustomQueries/server/spring-boot-jersey/src/main/java/com/server/reveal/DataSourceProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVDataSourceProvider;
import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.model.*;

import java.util.Objects;

public class DataSourceProvider implements IRVDataSourceProvider {
    public RVDataSourceItem changeDataSourceItem(IRVUserContext userContext, String dashboardsID, RVDataSourceItem dataSourceItem) {

        if (dataSourceItem instanceof RVSqlServerDataSourceItem sqlDataSourceItem) {
            //update underlying data source
            changeDataSource(userContext, dataSourceItem.getDataSource());

            //only change the table if we have selected our custom data source item
            if (Objects.equals(dataSourceItem.getId(), "MySqlServerDataSourceItem")) {
                //get the sales-person-id from the userContext
                var salesPersonId = userContext.getProperties().get("sales-person-id");

                //parametrize your custom query with the property obtained before
                sqlDataSourceItem.setCustomQuery(String.format("SELECT * FROM [Sales].[SalesOrderHeader] WHERE [SalesPersonId] = %s", salesPersonId));
            }
        }

        if (dataSourceItem instanceof RVMySqlDataSourceItem mySqlDataSourceItem) {
            //update underlying data source
            changeDataSource(userContext, dataSourceItem.getDataSource());

            //only change the table if we have selected our custom data source item
            if (Objects.equals(dataSourceItem.getId(), "MyMySqlDataSourceItem")) {
                //get the sales-person-id from the userContext
                var salesPersonId = userContext.getProperties().get("sales-person-id");

                //parametrize your custom query with the property obtained before
                //when using a MySqlDataSourceItem, you should use the fully qualified name for tables, otherwise it won't work
                mySqlDataSourceItem.setCustomQuery(String.format("SELECT * FROM AdventureWorks.Sales_SalesOrderHeader WHERE SalesPersonID = %s", salesPersonId));
            }
        }

        return dataSourceItem;
    }

    public RVDashboardDataSource changeDataSource(IRVUserContext userContext, RVDashboardDataSource dataSource) {
        if (dataSource instanceof RVSqlServerDataSource sqlDatasource) {
            sqlDatasource.setHost("your-host");
            sqlDatasource.setDatabase("your-database");
        }

        if (dataSource instanceof RVMySqlDataSource mySqlDataSource) {
            mySqlDataSource.setHost("your-host");
            mySqlDataSource.setDatabase("your-database");
        }

        return dataSource;
    }
}
